service/controller/clusterapi/v22/machine_deployment_resource_set.go

```diff
@@ -110,7 +112,7 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 			return false
 		}
 
-		if key.OperatorVersion(&cr) == VersionBundle().Version {
+		if key.OperatorVersion(&cr) == VersionBundle(config.Provider).Version {
```

Comment:
The calls here are ugly now but with flattening `cluster-operator` in the coming weeks we will change this because it will then simply be a comparison with `project.BundleVersion()`.

Reply:
How is it going to look like? `project.BundleVersion(config.Provider)`?

Reply:
Scratch that. I checked service/controller/clusterapi/v22/version_bundle.go and understand.